### PR TITLE
MEN-4665 sort parameter for get deployments endpoint

### PR DESCRIPTION
**Pull request by 0lmi**
https://tracker.mender.io/browse/MEN-4665

### CODE REVIEW

**Review by tranchitella — COMMENTED** [review]
A few notes:
- Use constants as much as possible, instead of having "asc" and "desc" you can define two constants SortDirectionAscending and SortDirectionDiscending in models and use them everywhere.
- The error in api_deployments.go is better defined as a var at the top of the file (es. ErrInvalidSortDirection) instead of being inline in thee code
- in `model/deployment.go`, please group non-stdlib imports together without new lines between them (they were already poorly sorted)
- extend unit tests (both API and the storage) to verify the sorting is applied correctly

**Comment by 0lmi** [comment]
@mender-test-bot start pipeline

File: api/http/api_deployments.go [hunk 1]
```diff
@@ -1194,6 +1197,15 @@ func ParseLookupQuery(vals url.Values) (model.Query, error) {
 		}
 	}
 
+	switch strings.ToLower(vals.Get("sort")) {
+	case model.SortDirectionAscending:
+		query.Sort = model.SortDirectionAscending
+	case "", model.SortDirectionDiscending:
+		query.Sort = model.SortDirectionDiscending
+	default:
+		return query, ErrInvalidSortDirection
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply api/http/api_deployments.go:1206]: oh sorry, I didn't pay attention to the return time, I thought it's a pointer.
Ignore this comment.

**Comment by 0lmi** [comment]
@tranchitella `test_demo_artifact` test is failing and it seems like it's known not related issue. Not sure if the PR can be approved and merged with this test failed. Leaving it on you to decide.

**Review by tranchitella — APPROVED** [review]

**Pull request merged**